### ISSUE TITLE
EVM BetaNet returns MainNet chain ID

### ISSUE THREAD

**Issue by artob** [near/nearcore#3781]
As discussed on our 2021-01-04 call, the EVM's <span class="error">[`CHAINID`]</span>([https://eips.ethereum.org/EIPS/eip-1344](https://eips.ethereum.org/EIPS/eip-1344)) opcode on betanet currently returns the registered value for mainnet:

*   [https://github.com/near/near-eth-rpc/blob/master/test/betanet.t](https://github.com/near/near-eth-rpc/blob/master/test/betanet.t)
*   [https://chainid.network](https://chainid.network)
*   [https://eips.ethereum.org/EIPS/eip-1344](https://eips.ethereum.org/EIPS/eip-1344)

**Comment by ilblackdragon** [comment]
This is a configuration issue on the betanet in the genesis.json

CC @chefsale

**Comment by chefsale** [comment]
I just checked the genesis.json hosted on S3 here: https://s3.console.aws.amazon.com/s3/object/build.nearprotocol.com?region=us-west-1&prefix=nearcore-deploy/betanet/genesis.json which nearup is using and the current betanet network RPC also returns betanet: http://rpc.betanet.near.org/status.

@artob are you sure you are have setup the node correctly? Can you provide more information on how did you reproduce this?

**Comment by chefsale** [comment]
@artob also if you are just running: `neard init --chain-id betanet` make sure you clean your ~/.near folder and try using `neard init --chain-id betanet --download-genesis` to get latest genesis download, as we also do a hardforks on betanet on protocol changes.

**Comment by ilblackdragon** [comment]
@chefsale Sorry for confusion, this issue is not about `chain_id` field but about new `evm_chain_id` field. `evm_chain_id` field should not be `1313161554` (which is MainNet `evm_chain_id`). We also registered `1313161555` for TestNet. So BetaNet prob should have `1313161556`.

**Comment by artob** [comment]
@chefsale Yes, as @ilblackdragon noted, this is about what the EVM's [`CHAINID`](https://eips.ethereum.org/EIPS/eip-1344) opcode returns. I'll edit the issue to make that clearer.

**Comment by chefsale** [comment]
It seems like the only place where we specify the chain ids for evm is here: https://github.com/near/nearcore/blob/master/core/chain-configs/src/genesis_config.rs, but we specify them for MAINNET_EVM_CHAIN_ID and TEST_EVM_CHAIN_ID, but we don't for betanet, so this seems like it's missing and is probably the cause for this. Just and assumption. cc: @artob

**Comment by chefsale** [comment]
And then here: https://github.com/near/nearcore/blob/master/neard/src/runtime.rs#L1397 we are not handling this properly, sending a fix.

**Comment by chefsale** [comment]
PR which I believe should be the fix: https://github.com/near/nearcore/pull/3800 cc: @mikedotexe @artob @ilblackdragon

**Comment by artob** [comment]
@chefsale Could you let me know when the changes from #3800 have been deployed to the betanet, please, and I'd go ahead and test that this has been resolved. (In general, do we have a dashboard that shows what betanet nodes are currently running?)

**Comment by chefsale** [comment]
@artob, sure they should be deployed in 20 minutes by the daily cron job.

**Comment by chefsale** [comment]
@artob the deploy has happened: https://buildkite.com/nearprotocol/deploy-betanet/builds/210. It was already done when you pinged me I just mixed up the time of the cron job.

**Comment by artob** [comment]
Turns out [near-web3-provider](https://github.com/near/near-web3-provider) has a bug regarding chain IDs (https://github.com/near/near-web3-provider/issues/71), so I'll need to test this by deploying an actual Solidity contract. I'll aim to come back to this tomorrow.

**Comment by bowenwang1996** [comment]
@artob what is the latest on this?
